src/net.cpp

```diff
@@ -2047,11 +2047,13 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
 
         int64_t nANow = GetAdjustedTime();
         int nTries = 0;
+        bool delete_anchors_file{false};
```

Comment:
If this code is added, I think the `delete_anchors_file` variable can be removed and the `DeleteAnchorsFile` in `CConnman::Start` can also be removed.

```suggestion
        if (m_anchors.empty()) DeleteAnchorsFile(gArgs.GetDataDirNet() / ANCHORS_DATABASE_FILENAME);
```